### PR TITLE
Add diagnostic monitoring APIs for DOM, VDM and status access for CMIS modules

### PR DESCRIPTION
**Pull request by mihirpat1**
#### Description
We need to add DOM, VDM and status related APIs to be inline with the changes proposed via the below HLD
[HLD for diagnostic monitoring of CMIS based transceivers by mihirpat1 · Pull Request #1828 · sonic-net/SONiC (github.com)](https://github.com/sonic-net/SONiC/pull/1828)

#### Motivation and Context
1. Moved definition of VDM freeze and unfreeze related APIs from c_cmis.py to cmis.py
2. Modified get_vdm_allpage to ensure that None is returned if module does not support VDM. Previously, the functions was checking for vdm group advertisement (page 2fh, byte 128.1:0) instead of checking for VDM pages supported (page 1h, byte 142.6)
3. Defined the following for CMIS transceivers
	a. get_transceiver_dom_flags
	b. get_transceiver_vdm_real_value
	c. get_transceiver_vdm_thresholds
	d. get_transceiver_vdm_flags
	e. get_transceiver_status_flags

#### How Has This Been Tested?

Tested the changes on the following types of transceivers
1. CMIS based transceiver
2. C-CMIS based transceiver
3. CMIS based transceivers without VDM advertised by the module

#### Additional Information (Optional)
MSFT ADO - 30215715

### CODE REVIEW

**Comment by mihirpat1** [comment]
@prgeor @qinchuanares Can you please help in reviewing this?

File: sonic_platform_base/sonic_xcvr/api/public/cmis.py [hunk 1]
```diff
@@ -2171,6 +2404,201 @@ def get_transceiver_loopback(self):
...
+        osnr{lane_num}                                 = FLOAT                  ; optical signal to noise ratio in db
+        esnr{lane_num}                                 = FLOAT                  ; electrical signal to noise ratio in db
+        cfo{lane_num}                                  = FLOAT                  ; carrier frequency offset in Hz
+        txcurrpower{lane_num}                          = FLOAT                  ; tx current output power in dbm
+        rxtotpower{lane_num}                           = FLOAT                  ; rx total power in  dbm
+        rxsigpower{lane_num}                           = FLOAT                  ; rx signal power in dbm
+        ========================================================================
+        """
+        vdm_real_value_dict = dict()
+        vdm_raw_dict = self.get_vdm(self.vdm.VDM_REAL_VALUE)
```
Comment (qinchuanares) [inline sonic_platform_base/sonic_xcvr/api/public/cmis.py:2460]: Shall we perform VDM freeze and unfreeze to update min/max/avg statistical values?

File: sonic_platform_base/sonic_xcvr/sfp_optoe_base.py [hunk 2]
```diff
@@ -47,6 +55,18 @@ def is_coherent_module(self):
         api = self.get_xcvr_api()
         return api.is_coherent_module() if api is not None else None
 
+    def get_transceiver_vdm_real_value(self):
```
Comment (prgeor) [inline sonic_platform_base/sonic_xcvr/sfp_optoe_base.py:58]: @mihirpat1 can you add comments what real means

File: sonic_platform_base/sonic_xcvr/api/public/cmis.py (hunk 1 above)
Comment (prgeor) [reply sonic_platform_base/sonic_xcvr/api/public/cmis.py:2460]: @mihirpat1 that is correct

**Review by prgeor — APPROVED** [review]

File: sonic_platform_base/sonic_xcvr/sfp_optoe_base.py (hunk 2 above)
Comment (mihirpat1) [reply sonic_platform_base/sonic_xcvr/sfp_optoe_base.py:58]: @prgeor I have addressed this now

**Review by prgeor — APPROVED** [review]

**Pull request merged**